syntax = "proto3";

message File {
    string fileName = 1; 
}

message singleFileResponse {
    string name = 1;
    string lastUpdated = 2;
    int32 size = 3;
}

message multipleFilesResponse {
    repeated singleFileResponse files = 1;
}

service fileService {
    rpc SearchFiles(File) returns (multipleFilesResponse);
    rpc ListAllFiles(Empty) returns (multipleFilesResponse);
}

message Empty {}

